phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixEmbeddedDriver.java

```diff
@@ -505,8 +519,21 @@ public ConnectionInfo(String zookeeperQuorum, Integer port, String rootNode, Str
             if (null == this.user) {
                 throw new RuntimeException("Acquired null user which should never happen");
             }
+            this.tenantID = tenantID;
         }
-        
+
+        public static String getTenantID(String url) {
+            if (url != null && url.contains(PhoenixRuntime.TENANT_ID_ATTRIB)) {
+                for (String param : url.split(";")) {
+                    if (param.startsWith(PhoenixRuntime.TENANT_ID_ATTRIB) &&
```

Comment:
I think this code may have some errors for a string like "TenantIdXY", is there protection against these type of non-matching?

Reply:
Could also have some weirdness for stuff like TenantId=TenantId=, might not be worth handling currently but the use of split and whether we allow ='s in the tenant Id.